spec/integration/thousand_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'open3'
+
+module RSpec::Core
+  RSpec.describe "Rspec", :slow do
+    def measure(cmd, *args)
+      Dir.mktmpdir do |dir|
+        yield dir
+        start = Time.now
+        path = File.absolute_path(Bundler.which(cmd))
+        Open3.popen3(path + ' ' + args.join(" "), chdir: dir) do |_, stdout, stderr|
+          stdout.read
+          stderr.read
+        end
+        return Time.now - start
+      end
+    end
+
+    # Calculates how many tests / second given tool can run
+    def benchmark(files, tests, delay)
+      minitest_timing = measure 'ruby', 'test.rb' do |dir|
+        File.write File.join(dir, 'test.rb'), """
+          $:<<'spec'  # add to load path
+          files = Dir.glob('spec/**/*.rb')
+          files.each{|file| require file.gsub(/^spec\\/|\\.rb$/,'')}
+        """
+
+        Dir.mkdir File.join(dir, 'spec')
+        files.times do |i|
+          File.write File.join(dir, 'spec', "test#{i}_spec.rb"), """
+            require 'minitest/autorun'
+            class TestMeme#{i} < Minitest::Test
+              def setup
+                sleep(#{delay/2})
+                @answer = 42
+              end
+          """ + (1..tests).map { |j| """
+              def test_#{j}
+                sleep(#{delay/2})
+                assert_equal @answer + #{j}, #{42 + j}
+              end
+          """ }.join("\n") + """
+            end
+          """
+        end
+      end
+
+      rspec_timing = measure('exe/rspec') do |dir|
```

Comment:
It might be better to have bin/rspec here

Reply:
I don't think bin/rspec works (there's no such file, at least on my machine). I think it's fine to measure exe/rspec because it's listed in gemspec